Loop/View Controllers/SettingsTableViewController.swift

```diff
@@ -378,100 +384,21 @@ final class SettingsTableViewController: UITableViewController, IdentifiableClas
     override func tableView(_ tableView: UITableView, shouldHighlightRowAt indexPath: IndexPath) -> Bool {
         return true
     }
-
+        
     override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
         let sender = tableView.cellForRow(at: indexPath)
 
         switch sections[indexPath.section] {
         case .pump:
             switch PumpRow(rawValue: indexPath.row)! {
             case .pumpSettings:
-                if var settings = dataManager.pumpManager?.settingsViewController() {
-                    settings.completionDelegate = self
-                    present(settings, animated: true)
+                pumpSelected {
```

Comment:
This code was copied as-is to a new function so that it could be re-used by the new UI ViewModel.